src/ert/_c_wrappers/enkf/ert_config.py

```diff
@@ -70,50 +74,94 @@ class ErtConfig:
     user_config_file: str = "no_config"
     config_path: str = field(init=False)
 
-    def __post_init__(self):
-        self.config_path = (
-            os.path.dirname(os.path.abspath(self.user_config_file))
-            if self.user_config_file
-            else os.getcwd()
-        )
-
     @classmethod
     def from_file(
-        cls, user_config_file, use_new_parser: bool = USE_NEW_PARSER_BY_DEFAULT
+        cls,
+        user_config_file,
+        use_new_parser: bool = USE_NEW_PARSER_BY_DEFAULT,
     ) -> "ErtConfig":
+        collected_errors: List[ErrorInfo] = []
+
         user_config_dict = ErtConfig.read_user_config(
             user_config_file, use_new_parser=use_new_parser
         )
+
         config_dir = os.path.abspath(os.path.dirname(user_config_file))
         ErtConfig._log_config_file(user_config_file)
         ErtConfig._log_config_dict(user_config_dict)
         ErtConfig.apply_config_content_defaults(user_config_dict, config_dir)
-        return ErtConfig.from_dict(user_config_dict)
+
+        the_config = ErtConfig.from_dict(user_config_dict, collected_errors)
+        ConfigValidationError.raise_from_collected(collected_errors)
+
+        return the_config
+
+    def __post_init__(self):
+        self.config_path = (
+            os.path.dirname(os.path.abspath(self.user_config_file))
+            if self.user_config_file
+            else os.getcwd()
+        )
 
     @classmethod
-    def from_dict(cls, config_dict) -> "ErtConfig":
+    def from_dict(
+        cls, config_dict, collected_errors: Optional[List[ErrorInfo]] = None
+    ) -> "ErtConfig":
+        do_raise_errors = False
+        if collected_errors is None:
+            collected_errors = []
+            do_raise_errors = True
+
         substitution_list = SubstitutionList.from_dict(config_dict=config_dict)
         config_dir = substitution_list.get("<CONFIG_PATH>", "")
         config_file = substitution_list.get("<CONFIG_FILE>", "no_config")
         config_file_path = os.path.join(config_dir, config_file)
 
-        ErtConfig._validate_dict(config_dict, config_file)
```

Comment:
We probably now desire to run as muc of the code as possible. Previously the code has been structured to stop early. I therefore propose that we ditch this _validate_dict, and let each part of the code that needs to detect error conditions which implies it cannot run, to detect and give an error in the list about that, and then simply skip calling that piece of the code, instead of skipping everything like before.